client/ayon_core/hosts/nuke/api/utils.py

```diff
@@ -103,33 +103,42 @@ def colorspace_exists_on_node(node, colorspace_name):
     except ValueError:
         # knob is not available on input node
         return False
-    all_clrs = get_colorspace_list(colorspace_knob)
 
-    return colorspace_name in all_clrs
+    return colorspace_name in get_colorspace_list(colorspace_knob)
 
 
 def get_colorspace_list(colorspace_knob):
     """Get available colorspace profile names
 
+    Because the values returned from colorspace_knob.values() do not correspond
+    to the value returned from colorspace_knob.value(), and the extracted
+    colorspace comes from using colorspace_knob.value(), we need to iterate
+    through all values to get the correct value.
```

Comment:
Oh boy - this really sounds like something that needs reporting to Nuke! :)

Reply:
Or does [`nuke.colorspaces.getColorspaceList(colorspace_knob)` work](https://learn.foundry.com/nuke/developers/100/pythonreference/nuke.colorspaces-module.html#getColorspaceList)?

Reply:
> What's the Knob class type? Is it an Array_Knob?
Its an `Enumeration_Knob`

> Does .array() work?
```
node = nuke.selectedNode()
print(node["colorspace"].array())
# Result: [0.0]
```
Does not work unfortunately.

> Or does [nuke.colorspaces.getColorspaceList(colorspace_knob) work](https://learn.foundry.com/nuke/developers/100/pythonreference/nuke.colorspaces-module.html#getColorspaceList)?

It looks like its the roles that are messing up the returned values.
```
import nuke


def get_colorspace_list(colorspace_knob):
    """Get available colorspace profile names

    Because the values returned from colorspace_knob.values() do not correspond
    to the value returned from colorspace_knob.value(), and the extracted
    colorspace comes from using colorspace_knob.value(), we need to iterate
    through all values to get the correct value.

    A code example of the above would be:

    for count, value in enumerate(colorspace_knob.values()):
        colorspace_knob.setValue(count)
        print(colorspace_knob.value() in colorspace_knob.values())

    Args:
        colorspace_knob (nuke.Knob): nuke knob object

    Returns:
        list: list of strings names of profiles
    """
    original_value = colorspace_knob.value()

    colorspaces = []

    try:
        for count, _ in enumerate(colorspace_knob.values()):
            colorspace_knob.setValue(count)
            colorspaces.append(colorspace_knob.value())
    finally:
        colorspace_knob.setValue(original_value)

    return colorspaces


node = nuke.selectedNode()
custom_colorspace_list = get_colorspace_list(node["colorspace"])
nuke_colorspace_list = nuke.getColorspaceList(node["colorspace"])
for count, value in enumerate(custom_colorspace_list):
    nuke_value = nuke_colorspace_list[count]
    if value != nuke_value:
        print(value)
        print(nuke_value)
```
```
get_colorspace_list:  color_picking
nuke.getColorspaceList:  color_picking	color_picking (Output - sRGB)
get_colorspace_list:  color_timing
nuke.getColorspaceList:  color_timing	color_timing (ACES - ACEScc)
get_colorspace_list:  compositing_linear
nuke.getColorspaceList:  compositing_linear	compositing_linear (ACES - ACEScg)
get_colorspace_list:  compositing_log
nuke.getColorspaceList:  compositing_log	compositing_log (Input - ADX - ADX10)
get_colorspace_list:  data
nuke.getColorspaceList:  data	data (Utility - Raw)
get_colorspace_list:  default
nuke.getColorspaceList:  default	default (ACES - ACES2065-1)
get_colorspace_list:  matte_paint
nuke.getColorspaceList:  matte_paint	matte_paint (Utility - sRGB - Texture)
get_colorspace_list:  reference
nuke.getColorspaceList:  reference	reference (Utility - Raw)
get_colorspace_list:  rendering
nuke.getColorspaceList:  rendering	rendering (ACES - ACEScg)
get_colorspace_list:  scene_linear
nuke.getColorspaceList:  scene_linear	scene_linear (ACES - ACEScg)
get_colorspace_list:  texture_paint
nuke.getColorspaceList:  texture_paint	texture_paint (ACES - ACEScc)
```
@jakubjezek001 got any idea why the roles would do this?

Reply:
This could be an alternative without the brute forcing value iteration, but there are a few assumptions in this method:
```
def get_colorspace_list_2(colorspace_knob):
    """Get available colorspace profile names

    Args:
        colorspace_knob (nuke.Knob): nuke knob object

    Returns:
        list: list of strings names of profiles
    """
    results = []
    pattern = r".*\t.* \(.*\)"
    for colorspace in nuke.getColorspaceList(colorspace_knob):
        # Regex of pattern matches roles
        match = re.search(pattern, colorspace)
        if match:
            results.append(colorspace.split("\t")[0])
        else:
            results.append(colorspace)

    return results
```
The assumptions being that the value returned from roles such as `texture_paint	texture_paint (ACES - ACEScc)` will be the first string before the tab.

Reply:
> Or does nuke.colorspaces.getColorspaceList(colorspace_knob) work?

I do agree with @BigRoy that it could be the fastest solution, even roles are adding some unnecessary text in items we could still search in its substrings.

So ideally rewrite the `ayon_core.hosts.nuke.api.utils.get_colorspace_list()` so it is rather wrapper of `nuke.colorspaces.getColorspaceList(colorspace_knob)` or even better just ditch it since there might not be any benefit from wrapper. 

Then in the function `ayon_core.hosts.nuke.api.utils.colorspace_exists_on_node()` refactor the check to be checking a substring of in the colorspaces list.  `any("scene_linear" in cl for cl in colorspaces)` will do and it will be much faster.

Reply:
Have used the code I proposed.